docs/release-notes/4420-substates-of-active

```diff
@@ -0,0 +1,11 @@
+:orphan:
+
+**API Changes**
+
+- The ``STATE_ACTIVE`` state for Experiments and Trials is divided into four sub-states:
+ ``STATE_QUEUED``, ``STATE_PULLING``, ``STATE_STARTING``, and ``STATE_RUNNING``.
+ Queries to ``GetExperimentsRequest`` which filter by state continue to use
+ ``STATE_ACTIVE``.
+
+- The state of Tasks is adjusted to match Experiments and Trials. Previous
+  ``STATE_PENDING`` and ``STATE_ASSIGNED`` become ``STATE_QUEUED``.
```

Comment:
Just a note, for context, assigned in k8s means pod created (not necessarily scheduled), in determined agents it means the task has acknowledged it received assignment (which actually means scheduled). Queued is sufficient since in determined agents nothing should stay in that start longer than an master:allocation<->agent:containers actor round trip, but just wanted to note.

Reply:
Could be a place for us, later, to internally improve the meaning of states too. So people dont have to know every RM implementation to manage.